test/e2e/unmanaged_dns_test.go

```diff
@@ -240,21 +243,40 @@ func TestUnmanagedDNSToManagedDNSInternalIngressController(t *testing.T) {
 		t.Fatalf("failed to update ingresscontroller %s: %v", name, err)
 	}
 
+	oldUID := lbService.UID
 	if err := kclient.Delete(context.TODO(), lbService); err != nil && !errors.IsNotFound(err) {
 		t.Fatalf("failed to delete svc %s: %v", lbService.Name, err)
 	}
 
+	// Ensure the service has been recreated
+	err := wait.PollImmediate(1*time.Second, 1*time.Minute, func() (bool, error) {
+		lbService := &corev1.Service{}
+		if err := kclient.Get(context.TODO(), controller.LoadBalancerServiceName(ic), lbService); err != nil {
+			t.Logf("Get %q failed: %v, retrying ...", controller.LoadBalancerServiceName(ic), err)
+			return false, nil
+		}
+		// Ensure the UID of the service changed.
+		if lbService.UID == oldUID {
+			t.Logf("Waiting for service %q to be deleted", controller.LoadBalancerServiceName(ic))
+			return false, nil
+		} else if ingresscontroller.IsServiceInternal(lbService) {
+			// The service got recreated, but is not external
+			t.Fatalf("load balancer %s is internal but should be external", lbService.Name)
```

Comment:
Shouldn't this just return an error? We catch and report any error after the call to PollImmediate.

Reply:
Andy's point is that if you returned an error here from the polling loop, then the `if err != nil { t.Fatalf(...) }` immediately after the loop would still suffice to terminate the test.

Reply:
Ah sorry I thought you meant `t.Errorf` vs. `t.Fatalf`.
Will fix in next push.